packages/veraswap-sdk/src/swap/getTransaction.ts

```diff
@@ -182,22 +187,36 @@ export async function getTransaction(
         case "BRIDGE": {
             const { currencyIn, currencyOut, amountIn, walletAddress, orbiterQuote, stargateQuote } = params;
             if (currencyIn.isNative && currencyOut.isNative) {
-                if (stargateQuote) {
-                    return getStargateETHBridgeTransaction({
-                        srcChain: currencyIn.chainId,
-                        dstChain: currencyOut.chainId,
-                        receiver: walletAddress,
-                        stargateQuote,
-                    });
+                if (!stargateQuote || stargateQuote.type !== "ETH") {
+                    if (!orbiterQuote) {
+                        throw new Error("Stargate ETH quote or Orbiter params are required for ETH bridging");
+                    }
+                    const { to, value, data } = orbiterQuote.steps[0].tx;
+                    return { to, value: BigInt(value), data };
                 }
 
-                if (!orbiterQuote) {
-                    throw new Error("Orbiter params are required for Orbiter bridging");
+                return getStargateETHBridgeTransaction({
+                    srcChain: currencyIn.chainId,
+                    dstChain: currencyOut.chainId,
+                    receiver: walletAddress,
+                    stargateQuote,
+                });
+            }
+
+            if (currencyIn.symbol && currencyIn.symbol in STARGATE_TOKEN_POOLS) {
+                if (!stargateQuote || stargateQuote.type !== "TOKEN") {
+                    throw new Error(`Stargate token quote is required for ${currencyIn.symbol} bridging`);
                 }
 
-                const { to, value, data } = orbiterQuote.steps[0].tx;
+                const tokenSymbol = currencyIn.symbol as keyof typeof STARGATE_TOKEN_POOLS;
 
-                return { to, value: BigInt(value), data };
+                return getStargateTokenBridgeTransaction({
```

Comment:
@leovigna do you think we should use a smart account for stargate token bridging, like hyperlane?
Else for now logic here seems to do a regular token approval.
Smart account would be cleaniner ig to avoid multiple transactions